### ISSUE TITLE
Event weighting for imbalanced datasets

### ISSUE THREAD

**Issue by annawoodard** [havakv/pycox#100]
Thanks for the great library! I'm training with an imbalanced dataset. In time-to-event prediction, is it reasonable to weigh positive examples, as can be done in the classification analog? If so, could a convenience keyword arg to the loss functions be added (see `pos_weight` [here](https://pytorch.org/docs/stable/generated/torch.nn.BCEWithLogitsLoss.html))? I think this would be similar to, but more efficient than, oversampling positive examples.

**Comment by havakv** [comment]
Thank you for the kind words!
This is a good question that I really can't answer well. From what I've seen in statistical survival analysis, there seems to be a lot of emphasis on calibrated estimates, in which case a weighted loss function would require re-weighting your predictions, so I don't think this have been explored to the extent that it probably should have been.
But if you only care about discrimination (e.g, concordance index) then I guess trying weighting should make sense.

I'm not sure I remember correctly (feel free to tell me I'm wrong here), but I think the weighting for linear regression and logistic regression is more of a numerical benefit, as mathematically the discriminative abilities should't really matter?  In that case, I would expect the same to hold true for the parametric survival methods in this package, and the way to verify if it makes sense is simply to test it.

Are there any specific methods you're thinking about here? The implementation probably differs a bit depending on the method, so I think it requires some changes
